dspback/routers/earthchem.py

```diff
@@ -158,7 +161,19 @@ async def delete_metadata_repository(self, request: Request, identifier):
         description="Creates a submission record of the EarthChem record.",
     )
     async def submit_repository_record(self, request: Request, identifier: str):
-        json_metadata = await self.submit(request, identifier)
+        try:
+            json_metadata = await self.submit(request, identifier)
+        except:
+            response = requests.get(
+                get_settings().earthchem_public_view_url % identifier,
+                headers={"accept": "application/json"},
+            )
+            if response.status_code >= 300:
+                raise
+
+            jsonld = scrape_jsonld(response.text, {"type": "application/ld+json"})
+            await create_or_update_submission_ecl_registration(self.user, jsonld, identifier)
+            return {"published": True}
```

Comment:
Is this response sufficient for the UI to display the metadata?

Reply:
The frontend does another get for the actual display.  This change is no longer needed now that the ECL API returns public resources.